dask/array/fft.py

```diff
@@ -0,0 +1,149 @@
+from __future__ import absolute_import
+
+import numpy as np
+import numpy.fft as npfft
+
+from .core import map_blocks
+
+
+chunk_error = ("Dask array only supports taking an FFT along an axis that \n"
+               "has a single chunk. An FFT operation was tried on axis %s \n"
+               "which has chunks %s. To change the array's chunks use "
+               "dask.Array.rechunk.")
+
+
+def _fft_wrap(fft_func, a, n, axis, dtype, chunks):
+    if len(a.chunks[axis]) != 1:
+        raise ValueError(chunk_error % (axis, a.chunks[axis]))
+
+    return map_blocks(lambda x: fft_func(x, n=n, axis=axis), a, dtype=dtype,
+                      chunks=chunks)
+
+
+def _fft_out_chunks(a, n, axis):
+    """ For computing the output chunks of fft and ifft"""
+    if n is None:
+        return a.chunks
+    chunks = list(a.chunks)
+    chunks[axis] = (n,)
+    return chunks
+
+
+def _rfft_out_chunks(a, n, axis):
+    if n is None:
+        n = a.chunks[axis][0]
+    chunks = list(a.chunks)
+    chunks[axis] = (n//2 + 1,)
+    return chunks
+
+
+def _irfft_out_chunks(a, n, axis):
+    if n is None:
+        n = 2 * (a.chunks[axis][0] - 1)
+    chunks = list(a.chunks)
+    chunks[axis] = (n,)
+    return chunks
+
+
+def _hfft_out_chunks(a, n, axis):
+    if n is None:
+        n = 2 * (a.chunks[axis][0] - 1)
+    chunks = list(a.chunks)
+    chunks[axis] = (n,)
+    return chunks
+
+
+def _ihfft_out_chunks(a, n, axis):
+    if n is None:
+        n = a.chunks[axis][0]
+    chunks = list(a.chunks)
+    if n % 2 == 0:
+        m = (n//2) + 1
+    else:
+        m = (n + 1)//2
+    chunks[axis] = (m,)
+    return chunks
+
+
+def fft(a, n=None, axis=-1):
+    """
+    Compute the one-dimensional discrete Fourier Transform along an axis that
+    only has one chunk.
+
+    The numpy.fft.fft docstring follows below:
+
+    """ + npfft.fft.__doc__
+
+    chunks = _fft_out_chunks(a, n, axis)
+
+    return _fft_wrap(npfft.fft, a, n, axis, np.complex_, chunks)
+
+
+def ifft(a, n=None, axis=-1):
+    """
+    Compute the one-dimensional inverse discrete Fourier Transform along an
+    axis that only has one chunk.
+
+    The numpy.fft.ifft docstring follows below:
+
+    """ + npfft.ifft.__doc__
+
+    chunks = _fft_out_chunks(a, n, axis)
+
+    return _fft_wrap(npfft.ifft, a, n, axis, np.complex_, chunks)
+
+
+def rfft(a, n=None, axis=-1):
+    """
+    Compute the one-dimensional discrete Fourier Transform for real input,
+    along an axis that has only one chunk.
+
+    The numpy.fft.rfft docstring follows below:
+
+    """ + npfft.rfft.__doc__
```

Comment:
Maybe we take the "no boiler plate" theme one step further and make our own wraps function inside of dask

```python
def wrap_preamble(preamble, func):
    def _(func):
        func.__doc__ = preamble + func.__doc__
        return func
    return _

preamble = """ Compute the one-dimensional ..."""

@wrap_preamble(preamble, rfft)
def rfft(a, n=None, axis=-1):
    chunks = ...
    return ...
```